aquarius/events/processors.py

```diff
@@ -352,3 +382,45 @@ def process(self):
         self.es_instance.update(self.asset, self.did)
 
         return self.asset
+
+
+class MetadataStateProcessor(EventProcessor):
+    def process(self):
+        self.did = make_did(self.event.address, self._chain_id)
+
+        if self.event.args.state != MetadataStates.ACTIVE:
+            try:
+                self.asset = self._es_instance.read(self.did)
+            except Exception:
+                self.asset = None
+            if self.asset and self.asset.get("id") == self.did:
+                self.soft_delete_ddo(self.did)
+            return True
+        else:
+            soft_deleted_ddo = self._es_instance.read(self.did)
+
+            receipt = self._web3.eth.get_transaction_receipt(
+                soft_deleted_ddo["event"]["tx"]
+            )
+
+            create_events = self.dt_contract.events[
+                EventTypes.EVENT_METADATA_CREATED
+            ]().processReceipt(receipt)
+            update_events = self.dt_contract.events[
+                EventTypes.EVENT_METADATA_UPDATED
+            ]().processReceipt(receipt)
+
+            event = create_events[0] if len(create_events) != 0 else update_events[0]
```

Comment:
if not create_events and not update_events:
(log the error and return False)

event = create_events[0] if create_events else update_events[0]